src/bt-core.c

```diff
@@ -70,7 +75,18 @@ init_bt_core(bt_callbacks_t* callbacks)
     goto err_bt_interface_init;
   }
 
+#if ANDROID_VERSION >= 21
+  status = bt_interface->set_os_callouts(callouts);
```

Comment:
Bluedroid simply crashes if these function pointers are not set properly. Bluedroid just blindly calls these functions without any initial value / nullptr checking.